src/messages/messages.controller.ts

```diff
@@ -2,14 +2,16 @@ import { Body, Controller, Delete, Get, Param, Post, Req, UseGuards } from '@nes
 import { Message } from './messages.entity';
 import { MessagesService } from './messages.service';
 import { AuthGuard } from '@nestjs/passport';
+import { EventsGateway } from './event.gateway';
 
 @UseGuards(AuthGuard('jwt'))
 @Controller('messages')
 export class MessagesController {
-  constructor(private service: MessagesService) {}
+  constructor(private service: MessagesService, private eventGateway: EventsGateway) {}
 
   @Post()
   create(@Body() message: Message, @Req() request) {
+    this.eventGateway.notify('[Chat] Chat Updated');
```

Comment:
You are about to notify clients before the message creation itself.
What if there would be an error inside `createMessage()`?

Reply:
@myownsumm Fixed. Is it correct now?